authors/apps/authentication/backends.py

```diff
@@ -1,10 +1,57 @@
-# import jwt
-#
-# from django.conf import settings
-#
-# from rest_framework import authentication, exceptions
-#
-# from .models import User
+import jwt
+from django.conf import settings
+from rest_framework import authentication, exceptions
+from .models import User
+from .serializers import LoginSerializer
+from .jwt_generator import jwt_decode
 
-"""Configure JWT Here"""
 
+class JWTAuthentication(authentication.BaseAuthentication):
+    def authenticate(self, request, username=None, password=None):
+        """
+        Returns a `User` if a correct username and password have been supplied
+        using HTTP Basic authentication.  Otherwise returns `None`.
+        """
+
+        auth_token = request.META.get('HTTP_AUTHORIZATION', b'').split()
+
+        if not auth_token or auth_token[0].lower() != 'bearer':
+            return None
+        if len(auth_token) == 1:
+            msg = ('Invalid basic header. No credentials provided.')
+            raise exceptions.AuthenticationFailed(msg)
+        elif len(auth_token) > 2:
+            msg = ('Invalid basic header.')
+            raise exceptions.AuthenticationFailed(msg)
+
+        try:
+            payload = jwt.decode(auth_token[1])
+        except:
+            msg = 'Invalid token. Please login'
+            raise exceptions.AuthenticationFailed(msg)
+
+        return self.authenticate_credentials(payload, password, request)
+
+    def authenticate_credentials(self, payload, password):
+        """
+        Authenticate the user.username and password against username and
+        password with optional request for context.
+        """
+
+        try:
+            user = User.objects.get(id=payload['user_id'])
+            valid_user = authenticate(
```

Comment:
It seems as though `authenticate` is not imported. 
Did this run on your machine? Based on the way that drf have implemented the same they have imported `authenticate`  as such `from django.contrib.auth import authenticate`

Reply:
This is actually a bug. The program is running fine because, at line 18, there is no token in the request since this is a login request and login request do not require a token hence the program exits at the return statement and the method `authenticate_credentials` is never called. I am working on it and will have a fix soon